packages/ui/src/machines/authenticator/__tests__/defaultService.test.ts

```diff
@@ -0,0 +1,105 @@
+import { PasswordSettings } from '../../../types';
+import { defaultServices } from '../defaultServices';
+
+const { validateFormPassword } = defaultServices;
+
+const untouched = { password: false };
+const touched = { password: true };
+
+const strictPasswordPolicy: PasswordSettings = {
+  passwordPolicyCharacters: [
+    'REQUIRES_LOWERCASE',
+    'REQUIRES_NUMBERS',
+    'REQUIRES_UPPERCASE',
+    'REQUIRES_SYMBOLS',
+  ],
+  passwordPolicyMinLength: 8,
+};
+
+const lenientPasswordPolicy: PasswordSettings = {
+  passwordPolicyCharacters: [],
+  passwordPolicyMinLength: 4,
+};
+
+describe('validateFormPassword', () => {
+  it('happy case with valid password and strict password policy', async () => {
+    const password = 'UnitTest_Password1';
+    const result = await validateFormPassword({ password }, touched, {
+      passwordPolicyCharacters: ['REQUIRES_SYMBOLS'],
+      passwordPolicyMinLength: 1,
+    });
+    expect(result).toBe(null);
+  });
+
+  it('unhappy cases with invalid passwords and strict password policy', async () => {
```

Comment:
Can we break this up in to three separate tests? Unless the results of a second test rely on the output of the first, tend to avoid grouping tests

Reply:
[1336cb0](https://github.com/aws-amplify/amplify-ui/pull/2071/commits/1336cb0747003cd4167359498ae19c5937da5458)